src/test/testing.java

```diff
@@ -0,0 +1,67 @@
+package test;
+
+import launch.PreauthLaunching;
+import org.openqa.selenium.*;
+import org.openqa.selenium.support.ui.Select;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import xpath.PreauthXPath;
+
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Properties;
+
+public class testing {
+    public WebDriver driver;
+    public Properties prop;
+    PreauthLaunching la=new PreauthLaunching();
+
+
+    @BeforeClass
+    void before_class() {
+        this.driver = la.driver;
+        this.prop = la.prop;
+    }
+
+    @Test
+    void login_valid() {
+        PreauthXPath obj2 = new PreauthXPath(driver);
+        driver.get("https://portal.gomobi.io/MobiversaAdmin/");
+        String Uname = prop.getProperty("uname");
+        obj2.username.sendKeys(Uname);
+        String Pass = prop.getProperty("pass");
+        obj2.password.sendKeys(Pass);
+        JavascriptExecutor jsExecutor = (JavascriptExecutor) driver;
+        jsExecutor.executeScript("arguments[0].click();", obj2.login);
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(6));
+        obj2.icon.click();
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(6));
+        jsExecutor.executeScript("arguments[0].click();", obj2.preauth);
+        obj2.from_date.click();
+        obj2.date1.click();
+        obj2.date_ok1.click();
+        obj2.to_date.click();
+        obj2.date2.click();
+        obj2.date_ok2.click();
+        obj2.date_search.click();
+    }
+    @Test
+        void validation(){
+            PreauthXPath obj2 = new PreauthXPath(driver);
+            JavascriptExecutor jsExecutor = (JavascriptExecutor) driver;
+            List<WebElement> rows=driver.findElements(By.xpath("//tbody/tr[@role=\"row\"]"));
```

Comment:
validation should be a  step in test ace, not a test method, all the test will have following steps,
1. test setup
2. test execution
3. test validation